sechub-wrapper-owasp-zap/src/test/java/com/mercedesbenz/sechub/zapwrapper/config/ZapWrapperDataSectionFileProviderTest.java

```diff
@@ -0,0 +1,366 @@
+
+    private ZapWrapperDataSectionFileProvider providerToTest = new ZapWrapperDataSectionFileProvider();;
+
+    @Test
+    void open_api_sources_folder_is_null_results_in_empty_list() {
+        /* execute */
+        List<File> apiDefinitionFiles = providerToTest.fetchApiDefinitionFiles(null, new SecHubScanConfiguration());
+
+        /* test */
+        assertTrue(apiDefinitionFiles.isEmpty());
+    }
+
+    @Test
+    void client_cert_sources_folder_is_null_results_in_null_as_cert_file() {
+        /* execute */
+        File certificateFile = providerToTest.fetchClientCertificateFile(null, new SecHubScanConfiguration());
+
+        /* test */
+        assertNull(certificateFile);
+    }
+
+    @Test
+    void open_api_sechub_scan_config_is_null_results_in_empty_list() {
+        /* execute */
+        List<File> apiDefinitionFiles = providerToTest.fetchApiDefinitionFiles("/example/path/to/extracted/sources", null);
+
+        /* test */
+        assertTrue(apiDefinitionFiles.isEmpty());
+    }
+
+    @Test
+    void client_cert_sechub_scan_config_is_null_results_in_null_as_cert_file() {
+        /* execute */
+        File certificateFile = providerToTest.fetchClientCertificateFile("/example/path/to/extracted/sources", null);
+
+        /* test */
+        assertNull(certificateFile);
+    }
+
+    @Test
+    void open_api_missing_data_section_part_results_in_empty_list() {
+        /* prepare */
+        String openApiSechubScanConfigJSON = """
+                {"apiVersion":"1.0","webScan":{"url":"https://localhost:8443","api":{"type":"openApi","use":["open-api-file-reference"]}}}""";
+        SecHubScanConfiguration openApiSechubScanConfiguration = SecHubScanConfiguration.createFromJSON(openApiSechubScanConfigJSON);
+
+        /* execute */
+        List<File> apiDefinitionFiles = providerToTest.fetchApiDefinitionFiles("/example/path/to/extracted/sources", openApiSechubScanConfiguration);
+
+        /* test */
+        assertTrue(apiDefinitionFiles.isEmpty());
+    }
+
+    @Test
+    void client_cert_missing_data_section_part_results_in_null_as_cert_file() {
+        /* prepare */
+        String clientCertSechubScanConfigJSON = """
+                {"apiVersion":"1.0","webScan":{"url":"https://localhost:8443","clientCertificate":{"password":"secret","use":["client-cert-file-reference"]}}}""";
+        SecHubScanConfiguration clientCertSechubScanConfiguration = SecHubScanConfiguration.createFromJSON(clientCertSechubScanConfigJSON);
+
+        /* execute */
+        File certificateFile = providerToTest.fetchClientCertificateFile("/example/path/to/extracted/sources", clientCertSechubScanConfiguration);
+
+        /* test */
+        assertNull(certificateFile);
+    }
+
+    @Test
+    void open_api_empty_sources_section_in_sechub_configuration_results_in_empty_list() {
+        /* prepare */
+        String openApiSechubScanConfigJSON = """
+                {"apiVersion":"1.0","data":{"sources":[]},
```

Comment:
Improve readability by using pretty printed JSON formatting, because it takes a while to understand the entire JSON structure without the formatting (difficult to read). This statement applies to all JSON snippets.

```suggestion
                {
                  "apiVersion":"1.0",
                  "data":{
                    "sources":[]
                  },
```